### PR TITLE
Implement Ford-Bellman algorithm and write tests for it

### PR DESCRIPTION
**Pull request by shakareem**
Implement Ford-Bellman algo for directed graphs.

Add tests for the algo, add util setup method.

### CODE REVIEW

File: app/src/main/kotlin/model/WeightedDirectedGraph.kt [hunk 1]
```diff
@@ -78,4 +78,60 @@ class WeightedDirectedGraph<D> : DirectedGraph<D>() {
...
+    // returns null if path doesn't exist
+    fun findShortestPathFordBellman(srcVertex: Vertex<D>, destVertex: Vertex<D>): List<Pair<Edge<D>, Vertex<D>>>? {
+        val NEG_INF = -1000000
+
+        val distance = MutableList(vertices.size) { Int.MAX_VALUE }
+        val predecessor = MutableList<Vertex<D>?>(vertices.size) { null }
+
+        distance[srcVertex.id] = 0
+
+        for (i in 0..vertices.size - 1) {
```
Comment (spisladqo) [inline app/src/main/kotlin/model/WeightedDirectedGraph.kt:91]: Can change to `for (i in vertices.indices)`

**Review by spisladqo — APPROVED** [review]
I really like your code, it is beautiful. Great work!

File: app/src/main/kotlin/model/WeightedDirectedGraph.kt (hunk 1 above)
Comment (shakareem) [reply app/src/main/kotlin/model/WeightedDirectedGraph.kt:91]: It is specific for the algo to iterate exactly number of vertices - 1 times, my code emphasizes that, but thanks for the remark!!

**Review by qrutyy — APPROVED** [review]
LGTM!

**Pull request merged**